files/initramfs-tools/union-mount.j2

```diff
@@ -41,42 +60,48 @@ set_tmpfs_log_partition_size()
 
 remove_not_in_allowlist_files()
 {
-    image_dir=$1
-    allowlist_file=${rootmnt}/host/$image_dir/allowlist_paths.conf
-  
-    # Return if the secure_boot_enable option is not set
-    if ! (cat /proc/cmdline | grep -i -q "secure_boot_enable=[y1]"); then
-        return
-    fi
+    local allowlist_file="$1"
+    local targeted_dir="$2"
+    local allowlist_pattern_file=/tmp/allowlist_paths.pattern
 
     # Return if the allowlist file does not exist
     if ! test -f "${allowlist_file}"; then
         echo "The file ${allowlist_file} is missing, failed to mount rw folder." 1>&2
         exit 1
     fi
 
-    rw_dir=${rootmnt}/host/$image_dir/rw
-
     # Set the grep pattern file, remove the blank line in config file
-    allowlist_pattern_file=${rootmnt}/host/$image_dir/allowlist_paths.pattern
-    awk -v rw_dir="$rw_dir" 'NF {print rw_dir"/"$0"$"}' ${allowlist_file} > $allowlist_pattern_file
+    awk -v rw_dir="$targeted_dir" 'NF {print rw_dir"/"$0"$"}' ${allowlist_file} > $allowlist_pattern_file
 
     # Find the files in the rw folder, and remove the files not in the allowlist
-    find ${rw_dir} -type f | grep -v -f $allowlist_pattern_file | xargs /bin/rm -f
+    find ${targeted_dir} -type f | grep -v -f $allowlist_pattern_file | xargs /bin/rm -f
     rm -f $allowlist_pattern_file
 }
 
 ## Mount the overlay file system: rw layer over squashfs
 image_dir=$(cat /proc/cmdline | sed -e 's/.*loop=\(\S*\)\/.*/\1/')
-mkdir -p ${rootmnt}/host/$image_dir/rw
-mkdir -p ${rootmnt}/host/$image_dir/work
+rw_dir=${rootmnt}/host/$image_dir/rw
+work_dir=${rootmnt}/host/$image_dir/work
+mkdir -p "$rw_dir"
+mkdir -p "$work_dir"
+
+swi_path="${rootmnt}/host/$(sed -E 's/.*loop=([^ ]+).*/\1/' /proc/cmdline)" # for Aboot
```

Comment:
Move into the branches, so it will not impact non-aboot.
```
if $secureboot; then
    if [ "$bootloader" = "aboot" ]; then
```

Reply:
I will move `swi_path=` declaration there but will keep the variable enhancements for `rw_dir` and `work_dir` since these are harmless.